src/ftest.jl

```diff
@@ -57,75 +55,95 @@ this is an ANOVA, our null hypothesis is that `Result ~ 1` fits the data as well
 ```jldoctest
 julia> dat = DataFrame(Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2.],
                        Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2],
-                       Other=[1, 1, 2, 1, 2, 1, 3, 1, 1, 2, 2, 1]);
-
-julia> model = lm(@formula(Result ~ 1 + Treatment), dat);
+                       Other=categorical([1, 1, 2, 1, 2, 1, 3, 1, 1, 2, 2, 1]));
 
 julia> nullmodel = lm(@formula(Result ~ 1), dat);
 
-julia> bigmodel = lm(@formula(Result ~ 1 + Treatment + Other), dat);
-
-julia> ft = ftest(model.model, nullmodel.model)
-        Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F* p(>F)
-Model 1       10   3      0.1283         0.9603
-Model 2       11   2   -1 3.2292 -3.1008 0.0000 0.9603 241.6234 <1e-7
+julia> model = lm(@formula(Result ~ 1 + Treatment), dat);
 
+julia> bigmodel = lm(@formula(Result ~ 1 + Treatment + Other), dat);
 
-julia> ftest(bigmodel.model, model.model, nullmodel.model)
-        Res. DOF DOF ΔDOF    SSR    ΔSSR     R²    ΔR²       F*  p(>F)
-Model 1        9   4      0.1038         0.9678
-Model 2       10   3   -1 0.1283 -0.0245 0.9603 0.0076   2.1236 0.1790
-Model 3       11   2   -1 3.2292 -3.1008 0.0000 0.9603 241.6234  <1e-7
+julia> ftest(nullmodel.model, model.model)
+──────────────────────────────────────────────────────────────────────
+     Res. DOF  ΔDOF     SSR     ΔSSR       R²     ΔR²        F*  p(>F)
```

Comment:
Something to discuss is whether to print the residual degrees of freedom of the consumed degrees of freedom. Other implementations are inconsistent in that regard.

It could make more sense to print the consumed ones, as they give an indication about the complexity of the model, while the residual degrees of freedom reflect the sample size (which is common to all models). It also takes fewer characters in the heading (just "DOF"). Opinions?